pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/catalog/PmdCatalogRule.java

```diff
@@ -39,18 +65,53 @@ public PmdCatalogRule(Element element, PmdCatalogCategory category, String langu
 		this.sourceJar = category.getSourceJar();
 	}
 
+    public String getLanguage() {
+        return language;
+    }
+
 	String getFullName() {
 		return getCategoryPath() + "/" + getName();
 	}
 
-	String getName() {
+	public String getName() {
 		return name;
 	}
 
+    public boolean isXpath() {
```

Comment:
is this how a typical XPath rule looks like?

<rule name="MyXpathRule" ...>
  <properties>
    <property name="maxStatements" type="Integer" value="10" min="1" max="40"
              description="Max number of statements per method"/>
    <property name="xpath">
    <![CDATA[
      //MethodDeclaration/Block[count(//BlockStatement) > $maxStatements]
    ]]></property>
  </properties>
</rule>

Ref: https://docs.pmd-code.org/latest/pmd_userdocs_extending_defining_properties.html#for-xpath-rules

and the code block here is just validating wether the XPATH structured properly?

Reply:
Rules can be Java-based or XPath-based. The structure for an XPath rule is:
```
<rule name="MyName" class="net.sourceforge.pmd.lang.rule.XPathRule">
    <properties>
        <property name="xpath">
            <value>
            Some CDATA expression
            </value>
        </property>
    </properties>
</rule>
```
The code block here is checking for the chain of `<rule><properties><property name="xpath">`, since that's the definite indicator that it's an XPath rule.